curvefs/src/client/fuse_client.cpp

```diff
@@ -1119,6 +1119,56 @@ CURVEFS_ERROR FuseClient::FuseOpGetXattr(fuse_req_t req, fuse_ino_t ino,
     return ret;
 }
 
+CURVEFS_ERROR FuseClient::FuseOpListXattr(fuse_req_t req, fuse_ino_t ino,
+                            char *value, size_t size, size_t *realSize) {
+    LOG(INFO) << "FuseOpListXattr, ino: " << ino
+              << ", size = " << size;
+    std::shared_ptr<InodeWrapper> inodeWrapper;
+    CURVEFS_ERROR ret = inodeManager_->GetInode(ino, inodeWrapper);
+    if (ret != CURVEFS_ERROR::OK) {
+        LOG(ERROR) << "inodeManager get inode fail, ret = " << ret
+                   << ", inodeid = " << ino;
+        return ret;
+    }
+
+    Inode inode = inodeWrapper->GetInodeLocked();
+    // get xattr key
+    for (const auto &it : inode.xattr()) {
+        // +1 because, the format is key\0key\0
+        *realSize += it.first.length() + 1;
+    }
+
+    // add summary xattr key
+    if (inode.type() == FsFileType::TYPE_DIRECTORY) {
+        *realSize += strlen(XATTRRFILES) + 1;
```

Comment:
if enable SummaryInfo, will it duplicated with xattr  store in inode.xattr?

Reply:
> if enable SummaryInfo, will it duplicated with xattr stored in inode.xattr?

That will not happen. The curve.dir.r* don't stored in inode xattr and here only return the xattr key.